modules/loader/src/test/java/com/opengamma/strata/loader/csv/TradeCsvLoaderTest.java

```diff
@@ -2000,6 +2052,34 @@ public void test_load_genericSecurity() {
   }
 
   //-------------------------------------------------------------------------
+
+  @Test
+  public void test_FxSingleBarrierOption() throws IOException {
+    ResourceLocator file = ResourceLocator.of(
+        "classpath:com/opengamma/strata/loader/csv/fx_single_barrier_option_trade.csv");
+
+    // Test write
+    StringBuilder testBuilder = new StringBuilder(1024);
+    TradeCsvWriter.standard().write(
+        ImmutableList.of(
+            expectedFxSingleBarrierOptionWithRebate(),
+            expectedFxSingleBarrierOptionWithoutRebate()),
+        testBuilder);
+
+    StringBuilder sampleBuilder = new StringBuilder();
+    file.getCharSource().lines().forEach(l -> sampleBuilder.append(l).append("\n"));
```

Comment:
Using `l` as a variable name is a bad idea as it looks like `1`

This code looks wrong anyway. The `CharSource` can just be read to a string without worrying about line endings. To compare two files handling line endings on Windows, just use `.replace("r\n", "\n")` on each input.